services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java

```diff
@@ -274,6 +275,13 @@ private void checkClient() {
 
     }
 
+    private void checkContentType(HttpRequest request) {
+        String contentTypeHeader = request.getHttpHeaders().getRequestHeaders().getFirst("Content-Type");
```

Comment:
IMO, `HttpHeaders.CONTENT_TYPE` might be better instead of using "Content-Type".